### PR TITLE
Create alias functionality

### PR DESCRIPTION
**Pull request by vegaro**

### CODE REVIEW

File: Purchases/Classes/RCBackend.m [hunk 1]
```diff
@@ -110,6 +110,21 @@ - (void)handle:(NSInteger)statusCode
...
+{
+    NSError *responseError = nil;
+    
+    if (statusCode > 300) {
+        BOOL finishable = (statusCode < 500);
+        NSString *message = response[@"message"] ?: @"Unknown backend error.";
+        responseError = [self errorWithBackendMessage:message finishable:finishable];
+    }
+    
+    completion(responseError);
```
Comment (jeiting) [inline Purchases/Classes/RCBackend.m]: Gotta nil check this.

File: Purchases/Classes/Public/RCPurchases.m [hunk 2]
```diff
@@ -560,4 +560,22 @@ - (void)updateOriginalApplicationVersion
...
+
+- (void)createAlias:(NSString *)alias completion:(void (^)(NSError * _Nullable error))completion
+{
+    [self.backend createAliasForAppUserID:self.appUserID withNewAppUserID:alias completion:^(NSError * _Nullable error) {
+        if (error == nil) {
+            [self.userDefaults setObject:alias forKey:RCAppUserDefaultsKey];
+            self.appUserID = alias;
+            completion(nil);
+        } else {
+            completion(error);
```
Comment (jeiting) [inline Purchases/Classes/Public/RCPurchases.m]: You can just follow if (error == nil) with just:
``` 
if (completion != nil) 
{
  completion(error)
}
```

File: Purchases/Classes/Public/RCPurchases.h [hunk 3]
```diff
@@ -235,6 +235,30 @@ typedef NS_ENUM(NSInteger, RCAttributionNetwork) {
...
+/**
+ This function will identify the current user with an appUserID. Tipically this would be used after a log out to identify a new user without calling configure
+ @param appUserID The appUserID that should be linked to the currently user
+ */
+- (void)identify:(NSString * _Nullable)appUserID;
+
+/**
+ * Resets the Purchases client clearing the save appUserID. This will generate a random user id and save it in the cache.
+ */
+- (void) reset;
```
Comment (jeiting) [inline Purchases/Classes/Public/RCPurchases.h]: space

File: Purchases/Classes/Public/RCPurchases.h [hunk 4]
```diff
@@ -235,6 +235,30 @@ typedef NS_ENUM(NSInteger, RCAttributionNetwork) {
...
+
+/**
+ This function will alias two appUserIDs together.
+ @param alias The new appUserID that should be linked to the currently identified appUserID
+ @param completion An optional completion block called when the aliasing has been successful. This completion block will receive an error if there's been one.
+ */
+- (void)createAlias:(NSString *)alias completion:(void (^)(NSError * _Nullable error))completion;
+
+/**
+ This function will identify the current user with an appUserID. Tipically this would be used after a log out to identify a new user without calling configure
```
Comment (jeiting) [inline Purchases/Classes/Public/RCPurchases.h]: Typically*, logout*,

File: Purchases/Classes/Public/RCPurchases.h (hunk 3 above)
Comment (jeiting) [reply Purchases/Classes/Public/RCPurchases.h]: "clearing the saved appUserID"

File: Purchases/Classes/Public/RCPurchases.m [hunk 5]
```diff
@@ -560,4 +571,41 @@ - (void)updateOriginalApplicationVersion
...
+        }
+    }];
+}
+
+- (void)identify:(NSString *)appUserID
+{
+    [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
+    self.appUserID = appUserID;
+    self.isUsingAnonymousID = NO;
```
Comment (jeiting) [inline Purchases/Classes/Public/RCPurchases.m]: We need to talk about `isUsingAnonymousID` here. Some people will want to use it with an identified user. 

We might want to change the name of it.

**Comment by vegaro** [comment]
@jeiting can I merge this?

**Pull request merged**